### ISSUE TITLE
Argument swaping

### ISSUE THREAD

**Issue by ElMehdiBouamama** [PacktPublishing/TensorFlow-Machine-Learning-Cookbook#5]
https://github.com/PacktPublishing/TensorFlow-Machine-Learning-Cookbook/blob/32140611e1f9e3c8a04040df7ace10cca16e0e2d/Chapter%2007/doc2vec.py#L115

The arguments of NCE loss should be swapped due to library updates.

**Comment by rtrevinnoc** [comment]
Uhmm... at first I tought that too. Now I think we were wrong.

From [TensorFlow r1.10](https://www.tensorflow.org/api_docs/python/tf/nn/nce_loss): `tf.nn.nce_loss(weights, biases, labels, inputs, num_sampled, num_classes, num_true=1, sampled_values=None, remove_accidental_hits=False, partition_strategy='mod', name='nce_loss')`

From [TensorFlow r1.5](https://www.tensorflow.org/versions/r1.5/api_docs/python/tf/nn/nce_loss): `tf.nn.nce_loss(
    weights,
    biases,
    labels,
    inputs,
    num_sampled,
    num_classes,
    num_true=1,
    sampled_values=None,
    remove_accidental_hits=False,
    partition_strategy='mod',
    name='nce_loss'
)`

From [TensorFlow r1.1](https://www.tensorflow.org/versions/r1.0/api_docs/python/tf/nn/nce_loss): `tf.nn.nce_loss(weights, biases, labels, inputs, num_sampled, num_classes, num_true=1, sampled_values=None, remove_accidental_hits=False, partition_strategy='mod', name='nce_loss')`

The order of the arguments are the same as you see, you can check them by yourself on the links above. Still, I am struggling to get this doc2vec example working on TF r1.10, if you can do it, please give me a call. I hope we can work together to accomplish it.

**Comment by MurtazaTinwala** [comment]
Hi @majomuchi123 

TensorFlow Machine Learning Cookbook - Second Edition is out now. 
Please have a look for all the updated content and codes in the new book on these links: 

Packt website - https://www.packtpub.com/big-data-and-business-intelligence/tensorflow-machine-learning-cookbook-second-edition

Amazon - https://www.amazon.com/dp/1789131685
